numpy/core/src/umath/ufunc_object.c

```diff
@@ -2066,41 +2126,63 @@ _parse_axes_arg(PyUFuncObject *ufunc, int core_num_dims[], PyObject *axes,
  */
 static int
 _get_coredim_sizes(PyUFuncObject *ufunc, PyArrayObject **op,
+                   int *core_num_dims, int * flexible_activated,
                    npy_intp* core_dim_sizes, int **remap_axis) {
     int i;
     int nin = ufunc->nin;
     int nout = ufunc->nout;
     int nop = nin + nout;
 
     for (i = 0; i < ufunc->core_num_dim_ix; ++i) {
-        core_dim_sizes[i] = -1;
+        /* support fixed-size dim names */
+        core_dim_sizes[i] = ufunc->core_dim_szs[i];
     }
     for (i = 0; i < nop; ++i) {
         if (op[i] != NULL) {
             int idim;
             int dim_offset = ufunc->core_offsets[i];
-            int num_dims = ufunc->core_num_dims[i];
+            int num_dims = core_num_dims[i];
             int core_start_dim = PyArray_NDIM(op[i]) - num_dims;
+            int dim_delta = 0;
+
+            /* Check if operands have enough dimensions */
+            if (core_start_dim < 0) {
```

Comment:
By the time we get here, shouldn't this already guaranteed to be true? If so, you could still put in an assert just to be sure.

Reply:
removed the code, checked previous to this call